executor/grant.go

```diff
@@ -211,25 +245,93 @@ func initColumnPrivEntry(ctx sessionctx.Context, user string, host string, db st
 	return err
 }
 
-// grantPriv grants priv to user in s.Level scope.
-func (e *GrantExec) grantPriv(priv *ast.PrivElem, user *ast.UserSpec) error {
+// grantGlobalPriv grants priv to user in global scope.
+func (e *GrantExec) grantGlobalPriv(user *ast.UserSpec) error {
+	if len(e.TLSOptions) == 0 {
+		return nil
+	}
+	priv, err := tlsOption2GlobalPriv(e.TLSOptions)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	sql := fmt.Sprintf(`UPDATE %s.%s SET PRIV = '%s' WHERE User='%s' AND Host='%s'`, mysql.SystemDB, mysql.GlobalPrivTable, priv, user.User.Username, user.User.Hostname)
+	_, _, err = e.ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
+	return err
+}
+
+var emptyGP = privileges.GlobalPrivValue{SSLType: privileges.SslTypeNotSpecified}
+
+func tlsOption2GlobalPriv(tlsOptions []*ast.TLSOption) (priv []byte, err error) {
+	if len(tlsOptions) == 0 {
+		priv = []byte("{}")
+		return
+	}
+	gp := privileges.GlobalPrivValue{SSLType: privileges.SslTypeNotSpecified}
+	for _, tlsOpt := range tlsOptions {
```

Comment:
Duplicate declare should not be allowed here, like;
`grant all on *.* to tester require cipher 'TLS_AES_256_GCM_SHA384' cipher 'RC4-SHA';`